api/extension-guides/ai/ai-extensibility-overview.md

```diff
@@ -0,0 +1,146 @@
+
+- Provide domain-specific expertise in ask mode
+- Control interaction flow and customize response behavior
+- Deep VS Code integration through extension APIs
+- Easy distribution and deployment via the Visual Studio Marketplace
+
+**Key considerations**:
+
+- Only available in ask mode
+- Remote deployment requires the extension to implement the client-server communication
+- Reuse across different tools requires modular design and implementation
+
+## Build your own AI-powered features
+
+VS Code gives you direct programmatic access to AI models for creating custom AI-powered features in your extensions. This approach enables you to build editor-specific interactions that use AI capabilities without relying on the chat interface.
+
+To use language models directly, use the [Language Model API](/api/extension-guides/ai/language-model) within your VS Code extension. You can incorporate these AI capabilities into any extension feature, such as code actions, hover providers, custom views, and more.
+
+**Key benefits**:
+
+- Create custom workflows and experiences that target specific use cases
+- Integrate AI capabilities into existing extension features
+- Deep VS Code integration through extension APIs
+- Easy distribution and deployment via the Visual Studio Marketplace
+
+**Key considerations**:
+
+- Reuse across chat or different tools requires modular design and implementation
+
+## Decide which option to use
+
+When choosing the right approach for extending AI in your VS Code extension, consider the following guidelines:
+
+1. **Choose Language Model Tool when**:
+    - You want to extend chat in VS Code with specialized capabilities
+    - You want automatic invocation based on user intent in agent mode
+    - You want access to VS Code APIs for deep integration in VS Code
+    - You want to distribute your tool through the VS Code Marketplace
+
+1. **Choose MCP Tool when**:
+    - You want to extend chat in VS Code with specialized capabilities
+    - You want automatic invocation based on user intent in agent mode
+    - You don't need to integrate with VS Code APIs
+    - Your tool needs to work across different environments (not just VS Code)
+    - Your tool should run remotely or locally
+
+1. **Choose Chat Participant when**:
+    - You want to extend ask mode with a specialized assistant with domain expertise
+    - You need to customize the entire interaction flow and response behavior
+    - You want access to VS Code APIs for deep integration in VS Code
+    - You want to distribute your tool through the VS Code Marketplace
+
+1. **Choose Language Model API when**:
+    - You want to integrate AI capabilities into existing extension features
+    - You're building UI experiences outside the chat interface
+    - You need direct programmatic control over AI model requests
+
+## Next steps
+
+Choose the approach that best fits your extension's goals:
+
+- [Implement a language model tool](/api/extension-guides/ai/tools)
```

Comment:
I think it is only fair to link to InlineCompletionsAPI (I hope we have a guide).
Since at the top you mention that completions can be extended which is true. And I think this needs to be a first class citizen in this doc.

Reply:
Link to API makes sense. And maybe just a sentence explaining it.

Reply:
Added "Extend code completions with the Inline Completions API"